### ISSUE TITLE
Remove direct reactor usage from twisted.conch.insults.window.TopWindow to make it more easily testable

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2413]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2413|
|Type|enhancement|
|Created|2007-01-21 14:15:37Z|

Currently TopWindow.repaint uses reactor.callLater, which makes unit testing inconvenient.  This should be indirected through a scheduling function which tests can replace to exert direct control over when things run.

<details><summary>Searchable metadata</summary>

```
trac-id__2413 2413
type__enhancement enhancement
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1169388937000000 1169388937000000
changetime__1169503276000000 1169503276000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Done in topwindow-scheduler-2413

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @exarkun|
|-|-|

Perfect. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [19440]) Merge topwindow-scheduler-2413

Author: exarkun
Reviewer: therve
Fixes [#2413](https://github.com/twisted/twisted/issues/2413)

Parameterize the scheduler function used by the insults TopWindow widget.
This allows unit tests to provide a deterministic scheduler so as to be
able to avoid having time-related race conditions.

This is a backwards incompatible change, introducing a new required argument
to the TopWindow initializer.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
